molecule/agent-install-test/prepare.yml

```diff
@@ -0,0 +1,17 @@
+---
+- name: Prepare
+  hosts: all
+  become: true
+
+  tasks:
+    - name: Create google auth directory
+      file:
+        path: /etc/google/auth
+        state: directory
+        mode: '0755'
+
+    - name: Copy application credentials onto the VM
```

Comment:
what is this for? What VM is this? I don't think the VM needs application credentials, does it?

Reply:
yes, this is required for authorizing the logging agent. https://cloud.google.com/logging/docs/agent/logging/authorization#copy-private-key

without this I get errors of the sort:  Failed to count metrics for uptime. error_class=Signet::AuthorizationError error=\"Error code 404 trying to get security access token\\nfrom Compute Engine metadata for the default service account. This\\nmay be because the virtual machine instance does not have permission\\nscopes specified.

Reply:
Weird. Have we grant the default compute engine service account of this project the following scopes: https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/authorization#before_you_begin?

If so, it should not require explicit application credential setup.

Which project are we using?

Reply:
I'm getting a 404 error when running `curl --silent --connect-timeout 1 -f -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/instance/service-accounts/default/scopes`

Locally, I'm using stackdriver-agents-rmoriar, whcih should have all the permissions outlined here https://g3doc.corp.google.com/cloud/monitoring/agents/g3doc/dev/service-accounts.md?cl=head#setup. But on korkoro we'll be using stackdriver-editor-serviceaccount-prod-key. 

We can try submitting this without these changes and see if kokoro runs into the same issues.

Reply:
Does molecule log the arguments it's passing to the VM creation request? I wonder if it's doing something funky like messing with the scopes specifically.

I'm fine with submitting and seeing if Kokoro hits problems, but I don't want to put us into a straightjacket where we have to use Kokoro to run the tests and nobody remembers why anymore. So ideally I'd like to understand the issue. (which is hard for me in this case since I have always been bad at authentication issues...)

Reply:
I have absolutely no idea, but I wonder if this part of one of the config files is related to this issue?

LoadPlugin exec
# Monitor the Stackdriver Logging agent. This should fail gracefully if for any
# reason the metrics endpoint for the Logging agent isn't reachable.
<Plugin "exec">
  # The script doesn't need any privileges, so run as 'nobody'.
  Exec "nobody" "/opt/stackdriver/collectd/bin/stackdriver-read_agent_logging" "http://localhost:24231/metrics"
</Plugin>

Reply:
> I have absolutely no idea, but I wonder if this part of one of the config files is related to this issue?
Unlikely, this config is copied over only for the monitoring agent and the install is failing for the logging agent.

> Does molecule log the arguments it's passing to the VM creation request? I wonder if it's doing something funky like messing with the scopes specifically.
Molecule invokes the [gcp_compute_instance ansible module](https://docs.ansible.com/ansible/latest/collections/google/cloud/gcp_compute_instance_module.html) which does take a scopes parameter and a service_accounts.scopes parameter however it's still failing after adding 
```
scopes:
  - https://www.googleapis.com/auth/logging.write
  - https://www.googleapis.com/auth/logging.admin
  - https://www.googleapis.com/auth/cloud-platform
```
I'll keep looking into it.

Reply:
I think this can be resolved by adding 
```
service_accounts:
  - scopes:
     - https://www.googleapis.com/auth/logging.write
     - https://www.googleapis.com/auth/logging.admin
     - https://www.googleapis.com/auth/cloud-platform
```

Now that the webhook is building and I have some feedback from the kokoro instance I can test this out.

Reply:
These scopes are necessary and sufficient for starting the agent on the VMs using Kokoro.